pymtl3/passes/adhoc_transform/AddDebugSignalPass.py

```diff
@@ -0,0 +1,42 @@
+"""
+========================================================================
+AddDebugSignalPass.py
+========================================================================
+TODO currently doesn't support array of components because the uniquification
+of added signal names of an array of components is very convoluted.
+The verilog translation support is limited for two components of the
+same class that have different names of debug signals.
+
+Author : Shunning Jiang
+Date   : Jul 5, 2020
+"""
+
+from pymtl3.dsl import Component, InPort, MetadataKey, OutPort
+from pymtl3.passes.BasePass import BasePass
+
+
+class AddDebugSignalPass( BasePass ):
+
+  debug_pins = MetadataKey(set)
+
+  def __call__( self, top, signal_names ):
```

Comment:
In this case you cannot `s.apply` the `AddDebugSignalPass` because it requires an extra argument to the `__call__` method... it would be nice if we can somehow make `apply` work here

Reply:
I agree. I didn't find a good solution